kvk-api/app.js

```diff
@@ -1,83 +1,126 @@
   try {
-    api_key = fs.readFileSync('/run/secrets/' + api_key_name, 'utf8');
-    console.log('API key succesfully loaded');
+    api_key = fs.readFileSync("/run/secrets/" + api_key_name, "utf8");
+    console.log("API key succesfully loaded");
   } catch (e) {
-    console.error('API key not found');
+    console.error("API key not found");
   }
 }
 
 async function fetchApi(endpoint, api_key) {
   var headers = {};
   if (api_key) {
     headers = {
-      'apikey': api_key
-    }
+      apikey: api_key,
+    };
   }
-  return fetch('https://' + endpoint, {
-      headers: headers
-    })
-    .catch(err => {
-      console.error('caught', err.message);
+  return fetch("https://" + endpoint, {
+    headers: headers,
+  })
+    .catch((err) => {
+      console.error("caught", err.message);
       return {
-        'error': 1
-      }
+        error: 1,
+      };
     })
-    .then(data => data.json())
+    .then((data) => data.json());
 }
 
 function numberCheck(req, res, next) {
   if (/^\d{8}$/.test(req.params.kvknr)) {
-    next()
+    next();
   } else {
-    res.sendStatus(400)
+    res.sendStatus(400);
   }
 }
 
+function transformData(kvkData) {
+  const {
+    registrationDate,
+    foundationDate,
+    employees,
+    businessActivities,
+    addresses,
+  } = kvkData.data.items[0];
+
+  const data = {
+    registrationDate,
+    foundationDate,
+    employees,
+  };
+
+  // TODO: Check there is exactly one main activity SBI code
+  const mainSbi = businessActivities
+    .filter((activity) => activity.isMainSbi)
+    .map((activity) => activity.sbiCode);
+
+  if (mainSbi.length > 0) {
+    data.mainSbi = mainSbi[0];
+  }
+
+  data.locatedInTheNetherlands =
+    addresses.filter(
+      (address) =>
+        address.type === "vestigingsadres" && address.country === "Nederland"
+    ).length > 0;
+
+  return data;
+}
+
 async function redisFetch(req, res, next) {
-  const kvknr = req.params.kvknr
-  const cacheval = await redis_kvk_cache.get(kvknr)
+  const kvknr = req.params.kvknr;
+  const cacheval = await redis_kvk_cache.get(kvknr);
   if (cacheval) {
-    res.json(JSON.parse(cacheval))
+    res.json(transformData(JSON.parse(cacheval)));
   } else {
-    next()
+    next();
   }
 }
 
-app.disable('x-powered-by');
+app.disable("x-powered-by");
 
-app.use('/', express.static('public'))
+app.use("/", express.static("public"));
 
-app.get('/api/companies/:kvknr', numberCheck, apiRateLimiter, redisFetch, async (req, res) => {
-  const kvknr = req.params.kvknr
-  const data = await fetchApi(url + endpoint + '?kvkNumber=' + kvknr, api_key)
-  await redis_kvk_cache.set(kvknr, JSON.stringify(data))
-  res.json(data)
-})
+app.get(
+  "/api/companies/:kvknr",
+  numberCheck,
+  apiRateLimiter,
+  redisFetch,
+  async (req, res) => {
+    const kvknr = req.params.kvknr;
+    const data = await fetchApi(
+      url + endpoint + "?kvkNumber=" + kvknr,
+      api_key
+    );
+    await redis_kvk_cache.set(kvknr, JSON.stringify(data));
```

Comment:
Transform data here

Reply:
My motivation for transforming the data every time are given at the top of this PR.